From f72ccac503c222e01f6774db3712979bba4a1962 Mon Sep 17 00:00:00 2001
From: Alfredo Moralejo <amoralej@redhat.com>
Date: Fri, 29 Sep 2017 18:55:22 +0200
Subject: [PATCH] Define key_manager/api_class to a cass full path

https://review.openstack.org/#/c/505752 has switched to use
key_manager.backend which is set by default to 'barbican'.
However, until fix in cinder [1] is merged, cinder expects it
to be a full class path. Castellan uses key_manager/api_class
to resolve key_manager.backend so, setting this parameter to
an existing class will fix jobs until [1] is merged.

A similar workaround has been pushed to p-o-i in [2]

[1] https://review.openstack.org/#/c/468252/
[2] https://review.openstack.org/#/c/507567/

Change-Id: I71755a9f5fb29116167c552988284903a850dfd8
---
 packstack/puppet/modules/packstack/manifests/cinder.pp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packstack/puppet/modules/packstack/manifests/cinder.pp b/packstack/puppet/modules/packstack/manifests/cinder.pp
index c7217b76..ab882860 100644
--- a/packstack/puppet/modules/packstack/manifests/cinder.pp
+++ b/packstack/puppet/modules/packstack/manifests/cinder.pp
@@ -37,6 +37,7 @@ class packstack::cinder ()
       nova_catalog_admin_info => 'compute:nova:adminURL',
       service_workers         => hiera('CONFIG_SERVICE_WORKERS'),
       default_volume_type     => $default_volume_type,
+      keymgr_api_class        => 'castellan.key_manager.barbican_key_manager.BarbicanKeyManager',
     }
 
     class { '::cinder::scheduler': }
-- 
2.13.5

